src/coreclr/jit/helperexpansion.cpp

```diff
@@ -0,0 +1,1043 @@
+        return result;
+    }
+
+    return fgExpandHelper<&Compiler::fgExpandThreadLocalAccessForCall>(true);
+}
+
+//------------------------------------------------------------------------------
+// fgExpandThreadLocalAccessForCall : Expand the CORINFO_HELP_GETSHARED_NONGCTHREADSTATIC_BASE_NOCTOR_OPTIMIZED
+//                             that access fields marked with [ThreadLocal].
+//
+// Arguments:
+//    block - Block containing the helper call to expand
+//    stmt  - Statement containing the helper call
+//    call  - The helper call
+//
+//
+// Returns:
+//    PhaseStatus indicating what, if anything, was changed.
+//
+// Notes:
+//    A cache is stored in thread local storage (TLS) of coreclr. It maps the typeIndex (embedded in
+//    the code at the JIT time) to the base of static blocks. This method generates code to
+//    extract the TLS, get the entry at which the cache is stored. Then it checks if the typeIndex of
+//    enclosing type of current field is present in the cache and if yes, extract out that can be directly
+//    accessed at the uses.
+//    If the entry is not present, the helper is called, which would make an entry of current static block
+//    in the cache.
+//
+bool Compiler::fgExpandThreadLocalAccessForCall(BasicBlock* block, Statement* stmt, GenTreeCall* call)
+{
+    assert(call->IsHelperCall());
+    if (!call->IsExpTLSFieldAccess())
+    {
+        return false;
+    }
+
+    CORINFO_THREAD_STATIC_BLOCKS_INFO threadStaticBlocksInfo;
+    info.compCompHnd->getThreadLocalStaticBlocksInfo(&threadStaticBlocksInfo);
+
+    assert(threadStaticBlocksInfo.tlsIndex.accessType == IAT_VALUE);
+    assert(eeGetHelperNum(call->gtCallMethHnd) == CORINFO_HELP_GETSHARED_NONGCTHREADSTATIC_BASE_NOCTOR_OPTIMIZED);
+
+    JITDUMP("Expanding thread static local access for [%06d] in " FMT_BB ":\n", dspTreeID(call), block->bbNum);
+    DISPTREE(call);
+    JITDUMP("\n");
+
+    call->ClearExpTLSFieldAccess();
+    assert(call->gtArgs.CountArgs() == 1);
+
+    // Split block right before the call tree
+    BasicBlock* prevBb       = block;
+    GenTree**   callUse      = nullptr;
+    Statement*  newFirstStmt = nullptr;
+    DebugInfo   debugInfo    = stmt->GetDebugInfo();
+    block                    = fgSplitBlockBeforeTree(block, stmt, call, &newFirstStmt, &callUse);
+    assert(prevBb != nullptr && block != nullptr);
+
+    // Block ops inserted by the split need to be morphed here since we are after morph.
+    // We cannot morph stmt yet as we may modify it further below, and the morphing
+    // could invalidate callUse.
+    while ((newFirstStmt != nullptr) && (newFirstStmt != stmt))
+    {
+        fgMorphStmtBlockOps(block, newFirstStmt);
+        newFirstStmt = newFirstStmt->GetNextStmt();
+    }
+
+    GenTreeLclVar* threadStaticBlockLcl = nullptr;
+
+    // Grab a temp to store result (it's assigned from either fastPathBb or fallbackBb)
+    if (threadStaticBlockLcl == nullptr)
```

Comment:
The variable was assigned nullptr, then immediately there's a `if (var == nullptr)`?